### PR TITLE
Task av/j131 delete episode functionality

### PR DESCRIPTION
**Pull request by artem-totality**

### CODE REVIEW

File: backend/src/api/episodes/episodes.api.ts [hunk 1]
```diff
@@ -69,6 +69,16 @@ const initEpisodesApi = ({ apiRouter, episodeService }: Args): Router => {
     }),
   );
 
+  episodeRouter.delete(
+    EpisodesApiPath.$ID,
+    checkAuthMiddleware(HttpMethod.DELETE),
```
Comment (ghost) [inline backend/src/api/episodes/episodes.api.ts:76]: what if the owner is not trying to do it?

File: backend/src/services/record/record.service.ts [hunk 2]
```diff
@@ -19,6 +19,18 @@ class Record {
...
+  public delete(id: number): Promise<TRecord> {
+    return this.#recordRepository.delete(id);
+  }
+
+  public async deleteByEpisodeId(id: number): Promise<void> {
+    const record = await this.#recordRepository.getByEpisodeId(id);
+
+    if (record) {
+      await this.delete(record.id);
+    }
```
Comment (ghost) [inline backend/src/services/record/record.service.ts]:
```suggestion
```
you should check it at the level above

File: frontend/src/components/configurate-episode/configurate-episode.tsx [hunk 3]
```diff
@@ -43,12 +47,21 @@ const ConfigurateEpisode: React.FC = () => {
...
-        : <CreateEpisodeForm
-          onSubmit={handleFormSubmit}
-          payload={mapEpisode} />
-      }
+      {isLoading ? (
+        <Loader />
+      ) : (
+        <>
+          <CreateEpisodeForm onSubmit={handleFormSubmit} payload={mapEpisode} />
+          {isEdit && (
```
Comment (ghost) [inline frontend/src/components/configurate-episode/configurate-episode.tsx]: it should be available only for the podcast owner

File: frontend/src/store/configurate-episode/actions.ts [hunk 4]
```diff
@@ -70,4 +70,13 @@ const loadEpisode = createAsyncThunk<Episode, number, AsyncThunkConfig>(
...
   });
 
-export { create, edit, loadEpisode };
+const deleteEpisode = createAsyncThunk<void, number, AsyncThunkConfig>(
+  ActionType.DELETE_EPISODE,
+  async (id, { extra }) => {
+    const { episodeApi, notificationService } = extra;
+    await episodeApi.delete(id);
+
+    notificationService.success(NotificationTitle.SUCCESS, NotificationMessage.EPISODE_DELETED);
```
Comment (ghost) [inline frontend/src/store/configurate-episode/actions.ts:86]: add redirect

File: frontend/src/store/configurate-episode/actions.ts [hunk 5]
```diff
@@ -70,4 +70,13 @@ const loadEpisode = createAsyncThunk<Episode, number, AsyncThunkConfig>(
     return episode;
   });
 
-export { create, edit, loadEpisode };
+const deleteEpisode = createAsyncThunk<void, number, AsyncThunkConfig>(
```
Comment (ghost) [inline frontend/src/store/configurate-episode/actions.ts]:
```suggestion
const delete = createAsyncThunk<void, number, AsyncThunkConfig>(
```

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by JuliaDzyuba — DISMISSED** [review]

File: frontend/src/store/configurate-episode/actions.ts (hunk 5 above)
Comment (artem-totality) [reply frontend/src/store/configurate-episode/actions.ts]: Can`t rename, becouse "delete" is native js function.

File: frontend/src/components/configurate-episode/configurate-episode.tsx [hunk 6]
```diff
@@ -43,12 +52,21 @@ const ConfigurateEpisode: React.FC = () => {
...
-          onSubmit={handleFormSubmit}
-          payload={mapEpisode} />
-      }
+      {isLoading ? (
+        <Loader />
+      ) : isEdit ? isEpisodeOwner ? (
+        <>
+          <CreateEpisodeForm onSubmit={handleFormSubmit} payload={mapEpisode} />
+          <Button
+            label="Delete"
```
Comment (ghost) [inline frontend/src/components/configurate-episode/configurate-episode.tsx]: this functionality should be on `episode` page

**Review by ghost — CHANGES_REQUESTED** [review]

File: backend/src/middlewares/check-user-owner/check-user-owner.middleware.ts [hunk 7]
```diff
@@ -1,24 +1,2 @@
-import { RequestHandler } from 'express';
```
Comment (what1s1ove) [inline backend/src/middlewares/check-user-owner/check-user-owner.middleware.ts]: delete this file

File: frontend/src/store/configurate-episode/actions.ts [hunk 8]
```diff
@@ -70,4 +77,14 @@ const loadEpisode = createAsyncThunk<Episode, number, AsyncThunkConfig>(
     return episode;
   });
 
-export { create, edit, loadEpisode };
+const deleteEpisode = createAsyncThunk<void, DeleteActionEpisodePayload, AsyncThunkConfig>(
```
Comment (what1s1ove) [inline frontend/src/store/configurate-episode/actions.ts:80]:
```suggestion
const delete = createAsyncThunk<void, DeleteActionEpisodePayload, AsyncThunkConfig>(
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/store/configurate-episode/actions.ts (hunk 8 above)
Comment (artem-totality) [reply frontend/src/store/configurate-episode/actions.ts:80]: Can`t rename, becouse "delete" is name of native js function. ('delete' is not allowed as a variable declaration name.ts(1389))

Comment (what1s1ove) [reply frontend/src/store/configurate-episode/actions.ts:80]: Good! 👍

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**